rustworkx/__init__.pyi

```diff
@@ -213,7 +222,281 @@ from .token_swapper import graph_token_swapper as graph_token_swapper
+    disconnected: bool = ...,
+) -> float: ...
+def adjacency_matrix(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    weight_fn: Callable[[_T], float] | None = ...,
+    default_weight: float = ...,
+    null_value: float = ...,
+) -> np.ndarray: ...
+def all_simple_paths(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    from_: int,
+    to: int,
+    min_depth: int | None = ...,
+    cutoff: int | None = ...,
+) -> list[list[int]]: ...
+def floyd_warshall(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    weight_fn: Callable[[_T], float] | None = ...,
+    default_weight: float = ...,
+    parallel_threshold: int = ...,
+) -> AllPairsPathLengthMapping: ...
+def floyd_warshall_numpy(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    weight_fn: Callable[[_T], float] | None = ...,
+    default_weight: float = ...,
+    parallel_threshold: int = ...,
+) -> np.ndarray: ...
+def floyd_warshall_successor_and_distance(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    /,
+    weight_fn: Callable[[_T], float] | None = ...,
+    default_weight: float | None = ...,
+    parallel_threshold: int | None = ...,
+) -> tuple[np.ndarray, np.ndarray]: ...
+def astar_shortest_path(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    node: int,
+    goal_fn: Callable[[_S], bool],
+    edge_cost_fn: Callable[[_T], float],
+    estimate_cost_fn: Callable[[_S], float],
+) -> NodeIndices: ...
+def dijkstra_shortest_paths(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    source: int,
+    target: int | None = ...,
+    weight_fn: Callable[[_T], float] | None = ...,
+    default_weight: float = ...,
+    as_undirected: bool = ...,
+) -> PathMapping: ...
+def has_path(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T], source: int, target: int, as_undirected: bool = ...
+) -> bool: ...
+def all_pairs_dijkstra_shortest_paths(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    edge_cost_fn: Callable[[_T], float] | None,
+) -> AllPairsPathMapping: ...
+def all_pairs_all_simple_paths(
+    graph: PyGraph | PyDiGraph,
+    min_depth: int | None = ...,
+    cutoff: int | None = ...,
+) -> AllPairsMultiplePathMapping: ...
+def all_pairs_dijkstra_path_lengths(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    edge_cost_fn: Callable[[_T], float] | None,
+) -> AllPairsPathLengthMapping: ...
+def dijkstra_shortest_path_lengths(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    node: int,
+    edge_cost_fn: Callable[[_T], float] | None,
+    goal: int | None = ...,
+) -> PathLengthMapping: ...
+def k_shortest_path_lengths(
+    graph: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    start: int,
+    k: int,
+    edge_cost: Callable[[_T], float],
+    goal: int | None = ...,
+) -> PathLengthMapping: ...
+def dfs_edges(graph: PyGraph[_S, _T] | PyDiGraph[_S, _T], source: int | None = ...) -> EdgeList: ...
+def is_isomorphic(
+    first: PyGraph[_S, _T] | PyDiGraph[_S, _T],
+    second: PyGraph[_S, _T] | PyDiGraph[_S, _T],
```

Comment:
Is there a way to indicate that these types must match? Because you can't have first be `PyGraph` and  second be a `PyDiGraph`?

Reply:
This file got even longer but `@overload` should take care of that